### PR TITLE
fix(server): Relay getting stuck when sentry goes down for a while

### PR DESCRIPTION
**Pull request by RaduW**
Fixes bug that was causing Relay to never fetch project configurations if Sentry went down for a while.

### CODE REVIEW

File: docs/architecture/project-configuration.md [hunk 1]
```diff
@@ -99,7 +99,7 @@ Relay does not do any sort of management of project states in Redis. From Relay'
 
 If Relay obtains the project state from Upstream it will **NOT** insert it in Redis. It is up to other, external systems, to manage project states and add/remove/refresh them in Redis.
 
-### Fetching the project state from Upstream 
+##Fetching the project state from Upstream
```
Comment (untitaker) [inline docs/architecture/project-configuration.md]:
```suggestion
## Fetching the project state from Upstream 
```

**Review by untitaker — APPROVED** [review]

File: docs/architecture/project-configuration.md [hunk 2]
```diff
@@ -108,11 +108,12 @@ Here's what happens in the Upstream actor
 **Legend**
 
 * ProjStateCh - Project State Channel
+* UpstreamProjSrc - Upstream Project Source
```
Comment (jan-auer) [inline docs/architecture/project-configuration.md:111]: Can we name these a little more human-readable, like `UpstreamSource` and `StateChannel`?

File: relay-server/src/actors/project_upstream.rs [hunk 3]
```diff
@@ -238,7 +237,22 @@ impl UpstreamProjectSource {
...
+                    // failures but the channels have been cleaned up because the requests
+                    // expired we need to reset the backoff so that the next request is not
+                    // simply ignored (by handle) and does a schedule_fetch().
+                    // Explanation 2: We use the backoff member for two purposes:
+                    //  -1 to schedule repeated fetch requests (at less and less frequent intervals)
+                    //  -2 as a flag to know if a fetch is already scheduled.
+                    // Resetting it in here signals that we don't have a backoff scheduled (either
+                    // because everything went fine or because all the requests have expired).
+                    // Next time a user wants a project it should schedule fetch requests.
+                    slf.backoff.reset();
```
Comment (jan-auer) [inline relay-server/src/actors/project_upstream.rs:256]: Can we remove the call to `self.backoff.reset();` in `FetchProjectState` now?

File: relay-server/src/actors/project_upstream.rs [hunk 4]
```diff
@@ -218,7 +218,6 @@ impl UpstreamProjectSource {
                                         Some(ProjectState::err())
                                     })
                                     .unwrap_or_else(ProjectState::missing);
-
```
Comment (jan-auer) [inline relay-server/src/actors/project_upstream.rs]: _nit_: Unintended change?

File: docs/architecture/project-configuration.md (hunk 2 above)
Comment (RaduW) [reply docs/architecture/project-configuration.md:111]: ok, fixed

File: relay-server/src/actors/project_upstream.rs (hunk 3 above)
Comment (RaduW) [reply relay-server/src/actors/project_upstream.rs:256]: I would like to keep `self.backoff.reset()` in `UpstreamProjectSource::handle()`.
Although not strictly necessary (at least I can't think of a case where we would make a difference) it doesn't hurt either and it makes it clear that we start a fetch cycle with a reseted backoff instance.

**Pull request merged**